modules/openapi-generator/src/main/resources/dart/libraries/dio/serialization/freezed/class.mustache

```diff
@@ -0,0 +1,47 @@
+part of 'models.dart';
+
+/// {{{description}}}{{^description}}{{classname}}{{/description}}
+{{#hasVars}}
+    ///
+    /// Properties:
+    {{#allVars}}
+        /// * [{{{name}}}] {{#description}}- {{{.}}}{{/description}}
+    {{/allVars}}
+{{/hasVars}}
+
+@freezed
+class {{classname}} with _${{classname}} {
+const {{classname}}._();
+
+
+{{>serialization/freezed/class_factory}}
+
+{{>serialization/freezed/class_from_json}}
+
+{{! toJson is only required for oneOf types}}
+{{#hasDiscriminatorWithNonEmptyMapping}}
```

Comment:
oneOf might not have a discriminator, e.g.: oneOf Primitives

Reply:
As far as I understood it is necessary to have a discriminator for both anyOf and oneOf cases.. at least from the docs..
https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/

Reply:
that's because you are looking in the inheritance section, so a discriminator becomes necessary, but take a look at [this](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) and you will see that they are optional.

Also, try to validate your implementation against these:

- [oneOf.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOf.yaml)
- [oneOfArrayMapImport.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOfArrayMapImport.yaml)
- [oneOfDiscriminator.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOfDiscriminator.yaml)
- [oneOf_nonPrimitive.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOf_nonPrimitive.yaml)
- [oneOf_primitive.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOf_primitive.yaml)
- [oneof_polymorphism_and_inheritance.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneof_polymorphism_and_inheritance.yaml)
- [anyOf.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/anyOf.yaml)
- [anyOfDiscriminator.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/anyOfDiscriminator.yaml)

Reply:
Thanks for the info.. but this would required us to implement a iterative try catch based approach where we try to deserialize the object into one of the schemas and on failure try the next one. Even with that approach we would just deserialize to the first one that matches, i.e, if two schemas share the same properties then the first one in the iterable will get priority without a descriminator. This doesn't make sense to me. Would really like to know how this is handled in other languages.

Reply:
Looks like this is how you have approached deserialization in your package.
https://github.com/Bdaya-Dev/dart-oneof/blob/main/one_of_serializer/lib/src/one_of_serializer_base.dart#L24-L60

Reply:
this is the difference between `oneOf` and `anyOf`.
- `oneOf` only allows one match
- `anyOf` allows multiple matches

also note that the iterative approach is used in the c# generator

Reply:
> that's because you are looking in the inheritance section, so a discriminator becomes necessary, but take a look at [this](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) and you will see that they are optional.
> 
> Also, try to validate your implementation against these:
> 
> * [oneOf.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOf.yaml)
> * [oneOfArrayMapImport.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOfArrayMapImport.yaml)
> * [oneOfDiscriminator.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOfDiscriminator.yaml)
> * [oneOf_nonPrimitive.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOf_nonPrimitive.yaml)
> * [oneOf_primitive.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneOf_primitive.yaml)
> * [oneof_polymorphism_and_inheritance.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/oneof_polymorphism_and_inheritance.yaml)
> * [anyOf.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/anyOf.yaml)
> * [anyOfDiscriminator.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/anyOfDiscriminator.yaml)

@ahmednfwela I am taking a look into this once again and trying to handle cases without discriminator. However one of the example here doesn't make sense to me or I am missing something! For e.g. the `one_of_primitive` case without a property name doesn't make sense to me. What should be the json request body in this case.?

For e.g. is an invalid json format!
```
{
32
}
```
Also how is this json created in built_value implementation. Could you give an example so I can test it locally.?